crosstool/wrapped_clang.cc

```diff
@@ -292,16 +295,54 @@ std::string GetCurrentDirectory() {
   return cwd;
 }
 
+std::string exec(std::string cmd) {
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
+    if (!pipe) {
+      std::cerr << "Error: failed to open pipe to '" << cmd << "'" << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+      result += buffer.data();
+    }
+    return result;
+}
+
+std::string GetToolchainPath(const std::string &toolchain_id) {
+  // NOTE: This requires all toolchains to contain a 'clang' executable. This
+  // is true today for custom Swift toolchains, but could change in the future.
+  std::string output = exec("xcrun --find clang --toolchain " + toolchain_id);
+
+  if (output.empty()) {
+    std::cerr << "Error: TOOLCHAINS was set to '" << toolchain_id
+      << "' but no toolchain with that ID was found" << std::endl;
+    exit(EXIT_FAILURE);
+  } else if (output.find("XcodeDefault.xctoolchain") != std::string::npos) {
```

Comment:
I think we know what DEVELOPER_DIR actually is, and can instead check for that? Not sure if that would actually be better in any way though.

Reply:
Ignore this. That would be more brittle.